synapse/lib/cell.py

```diff
@@ -2454,6 +2455,45 @@ def initSslCtx(self, certpath, keypath):
         sslctx.load_cert_chain(certpath, keypath)
         return sslctx
 
+    def _get_user_for_handler(self, handler: s_httpapi.Handler) -> dict:
```

Comment:
Could this be a method on the handler?  Doesn't look like it needs the cell

Reply:
not really a handler method - this is kind of an artifact of an evolving implementation which was aware of multiple ways we stored data before settle on just stashing them on the object and calling it a day.

We need to be attribute aware since there is nothing preventing a cell from serving up a tornado handler which isn't a subclass of our Handler base; therefore could be missing our attributes.

Reply:
I'm inlining this.